backend/database/migration/001_khouryclassroom.sql

```diff
@@ -1,43 +1,89 @@
-  student_gh_username VARCHAR(255) NOT NULL,
-  ta_gh_username VARCHAR(255),
-  completed BOOLEAN NOT NULL,
-  started BOOLEAN NOT NULL,
-  FOREIGN KEY (assignment_id) REFERENCES assignments(id)
-);
-
-CREATE TABLE IF NOT EXISTS due_dates (
-  id SERIAL PRIMARY KEY,
-  due TIMESTAMP DEFAULT NOW() NOT NULL,
-  student_assignment_id INTEGER NOT NULL,
-  FOREIGN KEY (student_assignment_id) REFERENCES student_assignments(id)
-);
-
-CREATE TABLE IF NOT EXISTS regrades (
-  id SERIAL PRIMARY KEY,
-  student_gh_username VARCHAR(255) NOT NULL,
-  ta_gh_username VARCHAR(255) NOT NULL,
-  due_date_id INTEGER NOT NULL,
-  FOREIGN KEY (due_date_id) REFERENCES due_dates(id)
+CREATE TABLE IF NOT EXISTS classrooms (
+    id SERIAL PRIMARY KEY,
+    classroom_name VARCHAR(255) NOT NULL,
+    org_id INTEGER NOT NULL,
+    org_name VARCHAR(255) NOT NULL
+);
+
+CREATE TABLE IF NOT EXISTS classroom_tokens (
+    token VARCHAR(255) PRIMARY KEY, 
+    expires_in TIMESTAMP NOT NULL,
+    classroom_id INTEGER NOT NULL,
+    FOREIGN KEY (classroom_id) REFERENCES classrooms(id)
+);
+
+CREATE TABLE IF NOT EXISTS user_to_classroom (
+    github_username VARCHAR(255) PRIMARY KEY, 
+    github_user_id INTEGER UNIQUE NOT NULL,
+    role VARCHAR(255) NOT NULL,
+    classroom_id INTEGER NOT NULL,
+    FOREIGN KEY (classroom_id) REFERENCES classrooms(id)
+);
+
+CREATE TABLE IF NOT EXISTS assignment_outlines (
+    id SERIAL PRIMARY KEY,
+    template_repo_owner VARCHAR(255) NOT NULL,
+    template_repo_id VARCHAR(255) NOT NULL,
+    created_date TIMESTAMP DEFAULT NOW(),
+    name VARCHAR(255) NOT NULL,
+    classroom_id INTEGER NOT NULL,
+    FOREIGN KEY (classroom_id) REFERENCES classrooms(id)
+);
+
+CREATE TABLE IF NOT EXISTS assignment_tokens (
+    token VARCHAR(255) PRIMARY KEY,
+    expires_in TIMESTAMP NOT NULL,
+    assignment_outline_id INTEGER NOT NULL,
+    FOREIGN KEY (assignment_outline_id) REFERENCES assignment_outlines(id)
+);
+
+CREATE TABLE IF NOT EXISTS rubric_items (
+    id SERIAL PRIMARY KEY,
+    assignment_outline_id INTEGER NOT NULL,
+    point_value INTEGER NOT NULL,
+    explanation VARCHAR(255) NOT NULL,
+    FOREIGN KEY (assignment_outline_id) REFERENCES assignment_outlines(id)
+);
+
+
+CREATE TABLE IF NOT EXISTS student_works (
+    id SERIAL PRIMARY KEY,
+    assignment_outline_id INTEGER NOT NULL,
+    repo_name VARCHAR(255),
+    due_date TIMESTAMP NOT NULL,
+    FOREIGN KEY (assignment_outline_id) REFERENCES assignment_outlines(id)
+
+);
+
+CREATE TABLE IF NOT EXISTS students_to_student_work (
+    github_user_id INTEGER NOT NULL,
+    student_work_id INTEGER NOT NULL,
+    FOREIGN KEY (github_user_id) REFERENCES user_to_classroom(github_user_id),
+    FOREIGN KEY (student_work_id) REFERENCES student_works(id)
+);
+
+CREATE TABLE IF NOT EXISTS submissions (
+    id SERIAL PRIMARY KEY, 
+    student_work_id INTEGER NOT NULL,
+    repo_name VARCHAR(255) NOT NULL,
+    grading_completed BOOLEAN DEFAULT FALSE NOT NULL,
```

Comment:
Instead of multiple boolean flags to represent the "state" of a submission, there should be one field that represents the state, which should be a string enumeration IMO. The reason for this is that it makes no sense that a submission could be published BUT not have grading completed.